### ISSUE TITLE
Trying to pass an array containing dictionaries as props to a StoryUiComponent with Mapping isn't working 

### ISSUE THREAD

**Issue by dsfai2020** [dsfai2020/Task-and-Nutrition-Manager#69]
**Describe the bug**
The data that was passed into the state object wasn't parsed before it was sent.  So what passed to my state object was a actually an array containing a STRING version of the dictionary.  The keys were properly created by cycling and filtering through the local storage using a criteria but when I assigned the values I didn't parse it.  

**To Reproduce**
1. Go to the browser press Control Shift i while the react app is running.  Press control R to refresh the page.
2. Click on console.
3. Scroll down to the bottom of the page and take a look at the UI that generated.  Also view the console logs.
4. See error that the UI is generating a STRING.  See that the way that the information was saved was as a string and not a dictionary.  It LOOKS like a dictionary but there are 'quotes' around it.

**Expected behavior**
What I would like have to happen is that based on the length of how many dictionaries are within my object, I'd like my Ui Component to render.  But I want to pass parameters that were saved from the local storage.  I accessed those parameters and filtered them into a single array containing keys and values.  I made it so that the key was actually the name of the ui as they were generated.  But I need the values to be accessible when I drill into the object further.  The reason is so that I can have the UI ultimately regenerate itself if there are records in the local storage already present.  I want the parameters to line up any changes that are made to those UI are actually happening to the right ones.  I don't want to create additional UI with partial data that isn't inheriting the props that are provided from the localstorage.

**Screenshots**
![image](https://github.com/dsfai2020/Task-and-Nutrition-Manager/assets/60169737/3395efc1-91d8-4f51-a307-c56e09826395)

**Desktop (please complete the following information):**

**Smartphone (please complete the following information):**

**Additional context**
Though I am able to access the initial keys of my objects the content within those keys aren't actually dictionaries.  So the entire string just gets pulled into it.  If you take a look at the picture you can see that the example I ran just rendered in a string dictionary over 3 iterations that I'd manually setup.  The assignments happened mostly in a useEffect and work around some trigger points that are in the picture.

**Comment by dsfai2020** [comment]
![image](https://github.com/dsfai2020/Task-and-Nutrition-Manager/assets/60169737/120687f8-1ecd-4569-9477-5622ab636cea)

**Comment by dsfai2020** [comment]
A lot of the code needed to be restructured.  I was using too many different variables to apply 'clones' of the data to in an attempt at being able to both test and capture the data from the local storage.  I went back into the original if statement and decided to work from here.  I created two simple a and b items.  A was for actually executing the getItem for command for the local storage.  Something that had gotten lost in the was further up the code.  I'd almost forgotten about it entirely and was trying to actually copy the data from an iteration of the local storage without actually grabbing it by the key.  I was able to combine my filter, the if statement that checked for a string inclusion of uiBackEnd, with the getItem command by simply referencing the each variable.  Being able to iterate through the local storage was just one part of the process.  Actually grabbing the information properly was another.
